processing/src/main/java/io/druid/query/QueryDimensionInfo.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.query;
+
+import io.druid.query.dimension.DimensionSpec;
+import io.druid.segment.ColumnValueSelector;
+import io.druid.segment.DimensionQueryHelper;
+
+/**
+ * A grouping of various related objects used during query processing for a single dimension, used for convenience.
+ *
+ * Each QueryDimensionInfo is associated with a single dimension.
+ */
+public class QueryDimensionInfo
+{
+  /**
+   * The DimensionSpec representing this QueryDimensionInfo's dimension, taken from the query being processed.
+   */
+  public final DimensionSpec spec;
```

Comment:
just out of curiosity, why not use getters instead of exposing these vars directly?